grasa_event_locator/views.py

```diff
@@ -480,29 +478,56 @@ def resetpw(request):
                         with connection.cursor() as cursor:
                                 cursor.execute("DELETE FROM `grasa_event_locator_resetpwurls` WHERE `user_ID` = '" + request.POST['emailAddr'] + "';")
                         resetlink = ''.join([random.choice(string.ascii_letters + string.digits) for n in range(15)])
-                        resetPWURL = resetPWURLs(user_ID = request.POST['emailAddr'], reset_string= resetlink)
+                        resetPWURL = resetPWURLs(user_ID = request.POST['emailAddr'], reset_string= resetlink, expiry_time= (datetime.now() + timedelta(minutes = 60)))
                         resetPWURL.save()
                         # Make sure to pull the hostname from config file.
-                        print(send_email([request.POST['emailAddr']], "GRASA - Reset Password", "You've requested a password reset at the GRASA Event Locator. Please visit this linnk: http://grasa.larrimore.de/resetPWForm/" + resetlink))
+                        print(send_email([request.POST['emailAddr']], "GRASA - Reset Password", "You've requested a password reset at the GRASA Event Locator. Please visit this link: http://grasa.larrimore.de/resetPWForm/" + resetlink + ". This link expires in 10 minutes."))
                 context = {'email_submitted': True}
                 return render(request, 'resetPW.html', context)
         else:
             return render(request, 'resetPW.html')
 
 def resetPWForm(request, reset_string):
-        if request.method == 'POST' and request.POST['new'] == request.POST['confirm']:
-                print(reset_string)
-                username = resetPWURLs.objects.get(reset_string=reset_string)
-                print(username)
-                user = User.objects.get(username=username.user_ID)
-                print(user)
-                user.set_password(request.POST['new'])
-                user.save()
+        try:
+            #Check if the current time is greater than the timestamp in the table (which is 10 minutes after submission)
+            if datetime.now() > datetime.strptime(resetPWURLs.objects.get(reset_string=reset_string).expiry_time, '%Y-%m-%d %H:%M:%S.%f'):
```

Comment:
is 10 minutes too short of a time window?

Reply:
Justin brought that up. A commit earlier changes it to hour, but the comment wasn't changed. Just fixed that now.